src/backends/policy_engine/souffle/datalog_lowering_visitor.h

```diff
@@ -30,13 +31,13 @@
 namespace raksha::backends::policy_engine::souffle {
 
 class DatalogLoweringVisitor
-    : public ir::IRTraversingVisitor<DatalogLoweringVisitor> {
+    : public ir::IRTraversingVisitor<DatalogLoweringVisitor, std::monostate> {
```

Comment:
The main problem with `void` is that it can't be passed as an argument to a function, except by pointer.
It makes things more complicated, but is possible, I just need to workout the template magic to make the ergonomics better.

Alternatively, I have a change which splits the return from the side effects so that we can have `void` functions and return values via different names. I had actually pushed it, but for some reason Github was showing the earlier change.

I'll investigate this.

Reply:
The main problem with `void` is that it can't be passed as an argument to a function, except by pointer.
It makes things more complicated, but is possible, I just need to workout the template magic to make the ergonomics better.

Alternatively, I have a change which splits the return from the side effects so that we can have `void` functions and return values via different names. I had actually pushed it, but for some reason GitHub was showing the earlier change.

I'll investigate this.

Reply:
I think this is resolved now with the backwards compatibility change and your approval in chat or `std::monostate` via the  `Unit` alias (added in `src/common/utils/types.h`.